product_variant_sale_price/models/product_product.py

```diff
@@ -34,20 +34,18 @@ def _compute_lst_price(self):
         uom_model = self.env["uom.uom"]
         for product in self:
             price = product.fix_price or product.list_price
-            if "uom" in self.env.context:
-                price = product.uom_id._compute_price(
-                    price, uom_model.browse(self.env.context["uom"])
-                )
+            if self.env.context.get("uom"):
+                context_uom = uom_model.browse(self.env.context.get("uom"))
+                price = product.uom_id._compute_price(price, context_uom)
             product.lst_price = price
 
     def _compute_list_price(self):
         uom_model = self.env["uom.uom"]
         for product in self:
             price = product.fix_price or product.product_tmpl_id.list_price
-            if "uom" in self.env.context:
-                price = product.uom_id._compute_price(
-                    price, uom_model.browse(self.env.context["uom"])
-                )
+            if self.env.context.get("uom"):
+                context_uom = uom_model.browse(self.env.context.get("uom"))
+                price = context_uom._compute_price(price, context_uom)
```

Comment:
```suggestion
                price = product.uom_id._compute_price(price, context_uom)
```
It should be like above?